### PR TITLE
feat: add support for magic string __CATEGORIES__ in query editor

### PR DESCRIPTION
**Pull request by iloveitaly**
This is super helpful for running custom queries in the editor

### CODE REVIEW

File: src/views/QueryExplorer.vue [hunk 1]
```diff
@@ -66,8 +68,19 @@ RETURN = sort_by_duration(merged_events);`,
   },
   methods: {
     query: async function () {
-      const query = this.query_code.split(';').map(s => s.trim() + ';');
+      let query = this.query_code
```
Comment (ErikBjare) [inline src/views/QueryExplorer.vue:71]:
```suggestion
      let query = this.query_code;
```

File: src/views/QueryExplorer.vue [hunk 2]
```diff
@@ -66,8 +68,19 @@ RETURN = sort_by_duration(merged_events);`,
...
+
+      // replace magic string `__CATEGORIES__` in query text with latest category rule
+      if(_.includes(query, '__CATEGORIES__')) {
```
Comment (ErikBjare) [inline src/views/QueryExplorer.vue:74]:
```suggestion
      if (_.includes(query, '__CATEGORIES__')) {
```

**Review by ErikBjare — COMMENTED** [review]
Nice! 

Can you just add a note to the Query view that this `__CATEGORIES__` thing exists? (preferably with an example)

Just so that it's somewhat discoverable and not an undocumented feature :)

**Comment by iloveitaly** [comment]
@ErikBjare done! Here's a PR https://github.com/ActivityWatch/docs/pull/98

File: src/views/QueryExplorer.vue [hunk 3]
```diff
@@ -53,6 +54,7 @@ afk_events = query_bucket(find_bucket("aw-watcher-afk_"));
 window_events = query_bucket(find_bucket("aw-watcher-window_"));
 window_events = filter_period_intersect(window_events, filter_keyvals(afk_events, "status", ["not-afk"]));
 merged_events = merge_events_by_keys(window_events, ["app", "title"]);
+merged_events = categorize(events, __CATEGORIES__);
```
Comment (ErikBjare) [inline src/views/QueryExplorer.vue:57]: Added this

**Comment by ErikBjare** [comment]
Merged, thanks for the contribution @iloveitaly ❤️

**Comment by cykirk** [comment]
This would be convenient functionality for outside of the query editor as well.

**Pull request merged**